transport/http_util.go

```diff
@@ -148,7 +149,7 @@ func isReservedHeader(hdr string) bool {
 // that should be propagated into metadata visible to users.
 func isWhitelistedPseudoHeader(hdr string) bool {
 	switch hdr {
-	case ":authority":
+	case ":authority", "user-agent":
```

Comment:
Why this change? "user-agent" isn't a pseudo header.

Reply:
To avoid [this](https://github.com/grpc/grpc-go/blob/7263089022e5a1615620b463eb70f4757697a873/transport/http_util.go#L342) condition so that `user-agent` can be added to metadata exposed to service handler.
Alternatively, we can add a case specifically for `user-agent` in the enclosing `select`. 
Or maybe even rename this function to be `isWhitelisted`instead.
Thoughts?

Reply:
Renaming the function to `isWhitelistedHeader` or similar seems good. Tweak the documentation for the function as well.